### PR TITLE
delete hardcoded handling of parameters

### PR DESCRIPTION
**Pull request by mtosi**
while trying to fix some issues of the HLT menu raised by the PR 7698
we realized some code in the release was defining w/in the c++ code some parameters (instead of making use of the config files !)

this PR fixes this issue

and it works regardless the integration of the PR 7698
runTheMatrix.py -l 1306.0 worked properly

### CODE REVIEW

**Comment by mtosi** [comment]
@perrotta @Martin-Grunewald @VinInn @rovere @fwyzard

File: RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc [hunk 1]
```diff
@@ -47,9 +47,13 @@ SeedFilter::SeedFilter(const edm::ParameterSet& conf,
   useZvertex_   = regionPSet.getParameter<bool>("useZInVertex");
   vertexSrc_    = tokens.token_vtx;
 
+  std::cout << "[SeedFilter::SeedFilter] conf: " << conf << std::endl;
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc]: cout's must be removed

File: RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc [hunk 2]
```diff
@@ -47,9 +47,13 @@ SeedFilter::SeedFilter(const edm::ParameterSet& conf,
...
+  
+
   // setup orderedhits setup (in order to tell seed generator to use pairs/triplets, which layers)
   edm::ParameterSet hitsfactoryPSet = conf.getParameter<edm::ParameterSet>("OrderedHitsFactoryPSet");
   std::string hitsfactoryName = hitsfactoryPSet.getParameter<std::string>("ComponentName");
+  std::cout << "[SeedFilter::SeedFilter] hitsfactoryPSet: " << hitsfactoryPSet << std::endl;
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc]: Likewise

File: RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc [hunk 3]
```diff
@@ -72,8 +76,15 @@ SeedFilter::SeedFilter(const edm::ParameterSet& conf,
   edm::ParameterSet creatorPSet;
   creatorPSet.addParameter<std::string>("propagator","PropagatorWithMaterial");
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc]: creatorPSet seems no longer be used - remove it completely

File: RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc [hunk 4]
```diff
@@ -72,8 +76,15 @@ SeedFilter::SeedFilter(const edm::ParameterSet& conf,
...
 
+  //  edm::ParameterSet seedCreatorPSet = hitsfactoryPSet.getParameter<edm::ParameterSet>("SeedCreatorPSet");
+  edm::ParameterSet seedCreatorPSet = conf.getParameter<edm::ParameterSet>("SeedCreatorPSet");
+  std::cout << "[SeedFilter::SeedFilter] seedCreatorPSet: " << seedCreatorPSet << std::endl;
+  std::string seedCreatorType = seedCreatorPSet.getParameter<std::string>("ComponentName");
+  std::cout << " ==> seedCreatorType: " << seedCreatorType << std::endl;
+
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc]: Remove cout's

File: RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc [hunk 5]
```diff
@@ -72,8 +76,15 @@ SeedFilter::SeedFilter(const edm::ParameterSet& conf,
...
   combinatorialSeedGenerator = new SeedGeneratorFromRegionHits(hitsGenerator,0,
-                                    SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator", creatorPSet)
+							       //                                    SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator", creatorPSet)
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronAlgos/src/SeedFilter.cc]: Remove commented line

File: RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc [hunk 6]
```diff
@@ -21,8 +21,19 @@ TSGFromOrderedHits::TSGFromOrderedHits(const edm::ParameterSet &pset,edm::Consum
 
   edm::ParameterSet creatorPSet;
   creatorPSet.addParameter<std::string>("propagator","PropagatorWithMaterial");
+  creatorPSet.addParameter<double>("SeedMomentumForBOFF",5.0);
```
Comment (Martin-Grunewald) [inline RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc]: Seem no longer be used - remove

File: RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc [hunk 7]
```diff
@@ -21,8 +21,19 @@ TSGFromOrderedHits::TSGFromOrderedHits(const edm::ParameterSet &pset,edm::Consum
...
+
+  std::cout << "[TSGFromOrderedHits::TSGFromOrderedHits] theConfig: " << theConfig << std::endl;
+
+  edm::ParameterSet seedCreatorPSet = theConfig.getParameter<edm::ParameterSet>("SeedCreatorPSet");
+  std::cout << "[TSGFromOrderedHits::TSGFromOrderedHits] seedCreatorPSet: " << seedCreatorPSet << std::endl;
+  std::string seedCreatorType = seedCreatorPSet.getParameter<std::string>("ComponentName");
+  std::cout << " ==> seedCreatorType: " << seedCreatorType << std::endl;
```
Comment (Martin-Grunewald) [inline RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc]: Remove all cout's

File: RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc [hunk 8]
```diff
@@ -21,8 +21,19 @@ TSGFromOrderedHits::TSGFromOrderedHits(const edm::ParameterSet &pset,edm::Consum
...
+
   theGenerator = new SeedGeneratorFromRegionHits(hitsGenerator, 0, 
-						 SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator", creatorPSet)
+						 //						 SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator", creatorPSet)
+						 //						 SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator", seedCreatorPSet)
+						 SeedCreatorFactory::get()->create(seedCreatorType, seedCreatorPSet)
 						 );
```
Comment (Martin-Grunewald) [inline RecoMuon/TrackerSeedGenerator/plugins/TSGFromOrderedHits.cc]: Remove commented lines

**Comment by Martin-Grunewald** [comment]
Hi Mia,

In general:
* no couts (MessageLogger is really needed)
* no commenting out of old/obsolete code - remove those lines (git keeps the record)
* remove unneeded variables

**Comment by mtosi** [comment]
yes, sorry for the inconvenience
they were there for the debugging, and I forgot to clean the code

it should be fine, now .. I hope

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedProducer.cc [hunk 9]
```diff
@@ -328,6 +328,17 @@ ElectronSeedProducer::fillDescriptions(edm::ConfigurationDescriptions& descripti
...
+
+    psd3.add<std::string>("ComponentName",std::string("SeedFromConsecutiveHitsCreator"));
+    psd3.add<std::string>("propagator",std::string("PropagatorWithMaterial"));
+    psd3.add<double>("SeedMomentumForBOFF",5.0);
+    psd3.add<double>("OriginTransverseErrorMultiplier",1.0);
+    psd3.add<double>("MinOneOverPtError",1.0);
+    psd3.add<std::string>("magneticField",std::string(""));
+    psd3.add<std::string>("TTRHBuilder",std::string("WithTrackAngle"));
+    psd3.add<bool>("forceKinematicWithRegionDirection",false);
+    psd0.add<edm::ParameterSetDescription>("SeedCreatorPSet",psd3);
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedProducer.cc]: Mia, what values should be put for the HLT for the parameters of the new SeedCreatorPSet in the ElectronSeedProducer?

Comment (mtosi) [inline RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedProducer.cc]: ciao martin

      SeedCreatorPSet = cms.PSet(
            ComponentName                     =
cms.string('SeedFromConsecutiveHitsCreator'),
            MinOneOverPtError                 = cms.double(1.0),
            OriginTransverseErrorMultiplier   = cms.double(1.0),
            SeedMomentumForBOFF               = cms.double(5.0),
*            TTRHBuilder                       =
cms.string('hltESPTTRHBWithTrackAngle'),*
            forceKinematicWithRegionDirection = cms.bool(False),
            magneticField                     = cms.string(''),
            propagator                        =
cms.string('PropagatorWithMaterial')
      ),

should be the config as the one currently used @HLT
thanks much
  mia

On Wed, Mar 18, 2015 at 10:36 PM, Martin Grunewald <notifications@github.com

**Comment by slava77** [comment]
-1

superseded by #8390

**Pull request closed without merging**